### PR TITLE
Use create/update to create or update container apps in source to cloud scenarios

### PR DESCRIPTION
**Pull request by snehapar9**
This change ensures container apps are created/updated using `create`/`update` command instead of `up` in source to cloud scenarios.

### CODE REVIEW

File: azurecontainerapps.ts [hunk 1]
```diff
@@ -109,7 +106,7 @@ export class azurecontainerapps {
     private static buildArguments: string;
     private static noIngressUpdate: boolean;
     private static useInternalRegistry: boolean;
-    private static shouldCreateOrUpdateContainerAppWithUp: boolean;
+    // private static shouldCreateOrUpdateContainerAppWithUp: boolean;
```
Comment (pauld-msft) [inline azurecontainerapps.ts:109]: remove

File: azurecontainerapps.ts [hunk 2]
```diff
@@ -653,16 +646,14 @@ export class azurecontainerapps {
...
                 await this.appHelper.updateContainerAppRegistryDetails(this.containerAppName, this.resourceGroup, this.registryUrl, this.registryUsername, this.registryPassword);
             }
 
             // Update the Container App using the 'update' command
             await this.appHelper.updateContainerApp(this.containerAppName, this.resourceGroup, this.commandLineArgs);
-        } else if (this.shouldCreateOrUpdateContainerAppWithUp) {
-            await this.appHelper.createOrUpdateContainerAppWithUp(this.containerAppName, this.resourceGroup, this.commandLineArgs);
         } else if (this.adminCredentialsProvided && !this.noIngressUpdate) {
             // Update the Container App with `up` command when admin credentials are provided and ingress is manually provided.
             await this.appHelper.updateContainerAppWithUp(this.containerAppName, this.resourceGroup, this.commandLineArgs, this.ingress, this.targetPort);
```
Comment (pauld-msft) [inline azurecontainerapps.ts:659]: we still appear to be updating with the Up command here

File: azurecontainerapps.ts [hunk 3]
```diff
@@ -621,11 +618,9 @@ export class azurecontainerapps {
         // Ensure '-i' argument and '--source' argument are not both provided
         if (!this.util.isNullOrEmpty(this.imageToDeploy)) {
             this.commandLineArgs.push(`-i ${this.imageToDeploy}`);
-        } else if (this.shouldCreateOrUpdateContainerAppWithUp) {
+        } else if (!this.util.isNullOrEmpty(this.appSourcePath) && this.useInternalRegistry) {
             this.commandLineArgs.push(`--source ${this.appSourcePath}`);
-            this.commandLineArgs.push(`-l ${this.location}`);
```
Comment (pauld-msft) [inline azurecontainerapps.ts:626]: is location arg not allowed / not working for `container app update`?

Comment (snehapar9) [reply azurecontainerapps.ts:626]: nope `update` does not support location. It derives it from env

File: azurecontainerapps.ts (hunk 2 above)
Comment (snehapar9) [reply azurecontainerapps.ts:659]: Yes, we still need these. This change just updates container app for only source to cloud scenarios. There are other scenarios that still require up and is already being handled

**Review by pauld-msft — APPROVED** [review]

File: azurecontainerapps.ts (hunk 1 above)
Comment (snehapar9) [reply azurecontainerapps.ts]: removed

**Review by harryli0108 — CHANGES_REQUESTED** [review]
LGTM :shipit:

**Review by harryli0108 — APPROVED** [review]

**Pull request merged**